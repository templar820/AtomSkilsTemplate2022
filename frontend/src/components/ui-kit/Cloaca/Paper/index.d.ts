import * as React from 'react';
import { EmotionStylesType } from 'lib-ui';

export interface PaperProps {
    /** Defines font-size */
    size?: string | number;
    /** Defines additional component css */
    css?: EmotionStylesType;
    /** Defines additional className */
    className?: string;
    /** Defines text or any node that will be placed in a body of the component */
    children?: React.ReactNode;
}

declare const Paper: React.ComponentType<PaperProps>;

export default Paper;
