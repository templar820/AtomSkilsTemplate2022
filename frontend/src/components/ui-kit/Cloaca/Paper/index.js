import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';

import { ColorsContext } from 'lib-root/colors';
import { hexToRGBA, autoAddPx } from 'lib-root/utils/styleMixins';
import { setClassName } from 'lib-root/utils/styleMixins';
import applyEmotionStyle from 'lib-root/utils/applyEmotionStyle';

const StyledPaper = styled('div', {
    shouldForwardProp: (prop) => isPropValid(prop) && prop !== 'size'
})`
    background-color: ${({ colors }) => colors.GrayScale_0};
    box-shadow: 0 0 ${({ colors, size }) => `${autoAddPx(size)} ${hexToRGBA(colors.GrayScale_700, 0.2)}`};
    ${({ css, ...restProps }) => applyEmotionStyle(css, restProps)};
`;

const Paper = React.forwardRef((props, ref) => {
    const colors = useContext(ColorsContext);
    return <StyledPaper {...{ colors, ref, ...props }} className={setClassName({ props: props, name: 'paper' })} />;
});

Paper.propTypes = {
    /** Defines font-size */
    size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Defines additional component css */
    css: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Defines additional className */
    className: PropTypes.string,
    /** Defines text or any node that will be placed in a body of the component */
    children: PropTypes.node
};
Paper.defaultProps = {
    size: '16px'
};

Paper.displayName = 'Paper';
export default Paper;
