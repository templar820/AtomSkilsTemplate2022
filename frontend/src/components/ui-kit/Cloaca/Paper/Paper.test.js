import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Paper from './';

Enzyme.configure({ adapter: new Adapter() });

const props = {
    size: 10,
    css: { background: 'brown' },
    className: 'toilet-paper',
    children: React.createElement(
        'legend',
        [],
        [
            <React.Fragment key="1">child</React.Fragment>,
            <React.Fragment key="2">
                <li>child</li>
            </React.Fragment>
        ]
    )
};

describe('<Paper />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Paper {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    // test must be first, otherwise the snapshot size will be too large
    test('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Paper {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('should render correctly', () => {
        expect(wrapper.prop('size')).toEqual(props.size);
        wrapper.setProps({ size: 'large' });
        expect(wrapper.prop('size')).toEqual('large');

        expect(wrapper.prop('css')).toMatchObject(props.css);
        expect(wrapper.hasClass(props.className)).toBeTruthy();
        expect(wrapper.contains(props.children)).toBeTruthy();
    });
});
