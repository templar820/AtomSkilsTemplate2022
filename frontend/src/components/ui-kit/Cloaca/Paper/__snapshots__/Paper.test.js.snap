// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paper /> should match snapshot 1`] = `
<StyledPaper
  className="toilet-paper CLR-paper"
  colors={
    Object {
      "GrayScale_0": "#ffffff",
      "GrayScale_100": "#EDEDED",
      "GrayScale_150": "#E5E5E5",
      "GrayScale_200": "#D8D8D8",
      "GrayScale_50": "#fafafa",
      "GrayScale_500": "#999999",
      "GrayScale_55": "#f6f6f6",
      "GrayScale_600": "#7C7C7C",
      "GrayScale_700": "#333333",
      "GrayScale_900": "#000000",
      "error": "#F62434",
      "errorAccent": "#EB5A40",
      "info": "#34AAF2",
      "infoAccent": "#5BD9E5",
      "primary": "#00B956",
      "primaryAccent": "#10E272",
      "secondary": "#731982",
      "secondaryAccent": "#A500BF",
      "tertiary": "#444189",
      "tertiaryAccent": "#404D46",
      "warning": "#FFA717",
      "warningAccent": "#F36F20",
    }
  }
  css={
    Object {
      "background": "brown",
    }
  }
  size={10}
>
  <legend>
    child
    <li>
      child
    </li>
  </legend>
</StyledPaper>
`;
